source/cl/tools/clc/clc.cpp

```diff
@@ -700,31 +700,34 @@ result driver::saveBinary() {
     binary = binary_storage;
   }
 
+  FILE *fp = nullptr;
+  bool owns_fp = false;
   if (output_file == "-") {
-    if (!dry_run) {
-      if (fwrite(binary.data(), 1, binary.size(), stdout) != binary.size()) {
-        perror("error: Could not write all of the binary to the output file");
+    fp = stdin;
+  } else if (verbose) {
+    (void)std::fprintf(stderr, "info: writing binary to %s\n",
+                       generated_output_file.c_str());
+  }
+  if (!dry_run) {
+    if (fp == nullptr) {
+      fp = fopen(generated_output_file.c_str(), "wb");
+      if (fp == nullptr) {
+        (void)perror("error: Could not open output file");
```

Comment:
I know this is not an addition in this patch, but do we need the `error: ` in the perror string? If not, might it be worth removing it?

Reply:
We are fairly consistent in prefixing messages with a severity (`info: ...`, `error: ...`), so I think we should keep it here.